webapp/src/main/java/com/box/l10n/mojito/rest/repository/BranchStatisticWS.java

```diff
@@ -73,15 +74,38 @@ public Page<BranchStatistic> getBranchesOfRepository(
                 .and(ifParamNotNull(deletedEquals(deleted)))
                 .and(ifParamNotNull(empty(empty)))
                 .and(ifParamNotNull(createdBefore(createdBefore)))
-                .and(ifParamNotNull(createdAfter(createdAfter)))
-                .and(totalCountLessThanOrEqualsTo(totalCountLte)),
+                .and(ifParamNotNull(createdAfter(createdAfter))),
             pageable);
 
+    // Retrieve the branch statistics where the total text unit count is greater than the limit,
+    // by filtering the branchStatisticIds less than the limit.
+    List<BranchStatistic> branchStatisticsWithoutTextUnits =
+        branchStatisticRepository.findAllGreaterThanById(
+            branchStatisticIds.getContent(), totalCountLte);
+
+    // Set the BranchTextUnitStatistics to null as the branch has a large number of text Units.
+    branchStatisticsWithoutTextUnits.forEach(
+        branchStatistic -> {
+          branchStatistic.setBranchTextUnitStatistics(null);
+        });
+
+    List<Long> branchStatisticIdsWithoutTextUnits =
+        branchStatisticsWithoutTextUnits.stream()
+            .map(BranchStatistic::getId)
+            .collect(Collectors.toList());
+
+    List<Long> branchStatisticIdsWithLessThan =
+        branchStatisticIds.getContent().stream()
+            .filter(i -> !branchStatisticIdsWithoutTextUnits.contains(i))
+            .collect(Collectors.toList());
+
     // 2. Hydrate BranchStatistic entities for JSON response using eager loading to avoid N+1
     // queries
     List<BranchStatistic> branchStatistics =
-        branchStatisticRepository.findByIdIn(branchStatisticIds.getContent(), pageable.getSort());
+        branchStatisticRepository.findByIdIn(branchStatisticIdsWithLessThan, pageable.getSort());
 
+    // Merge the two lists:
+    branchStatistics.addAll(branchStatisticsWithoutTextUnits);
```

Comment:
it seems that logic is changing the order of the results by putting them at the end of the page. Would that be an issue. I guess it is inside a page so it is not a big difference.

Reply:
Yes, it will be adding to the end of the page's end. I suppose we can order them as well, but not sure if it's necessary. Let me know if you want me to update it.

Reply:
yeah not sure it is super important, my initial thought is that we should keep the same order as what `Page<Long> branchStatisticIds =
        sparseBranchStatisticRepository.findAllWithIdOnly(` returns but probably not a big deal. Let's keep it like this